File: sem4/mpp/mpp-proiect-csharp-vreauladudu/services/IMotoService.cs
using mpp_proiect_csharp_vreauladudu.domain;
using System;
using System.Collections.Generic;

namespace mpp_proiect_csharp_vreauladudu.service
{
    public interface IMotoService
    {
        void login(User user, IMotoObserver client);
        void logout(User user, IMotoObserver client);
        Dictionary<int, List<Motorcycle>> listRaces(IMotoObserver client);
        IEnumerable<Motorcycle> listMotorcycles(IMotoObserver client);
        List<Motorcycle> listMotorcyclesByCapacity(int capacity, IMotoObserver client);
        List<Participant> findParticipantForRace(int motorcycleId, IMotoObserver client);
        List<Motorcycle> listMotorcyclesByBrand(String brand, IMotoObserver client);
        Motorcycle findMotorcycleById(int motorcycleId, IMotoObserver client);
        void addParticipant(String participantName, String motorcycleBrand, int motorcycleCapacity, IMotoObserver client);
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/services/src/main/java/moto/services/IMotoService.java
package moto.services;

import moto.Motorcycle;
import moto.Participant;
import moto.User;

import java.util.List;
import java.util.Map;

public interface IMotoService
{
    void login(User user, IMotoObserver client) throws MotoException;
    void logout(User user, IMotoObserver client) throws MotoException;
    Map<Integer, List<Motorcycle>> listRaces(IMotoObserver client) throws MotoException;
    Iterable<Motorcycle> listMotorcycles(IMotoObserver client) throws MotoException;
    List<Motorcycle> listMotorcyclesByCapacity(Integer capacity, IMotoObserver client) throws MotoException;
    List<Participant> findParticipantForRace(Integer motorcycleId, IMotoObserver client) throws MotoException;
    List<Motorcycle> listMotorcyclesByBrand(String brand, IMotoObserver client) throws MotoException;
    Motorcycle findMotorcycleById(Integer motorcycleId, IMotoObserver client) throws MotoException;
    void addParticipant(String participantName, String motorcycleBrand, Integer motorcycleCapacity, IMotoObserver client) throws MotoException;
}